clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/evaluation/EventAdapter.kt

```diff
@@ -110,22 +119,35 @@ class EventAdapter(
 
     @VisibleForTesting
     internal fun getActualPropertyValue(propertyName: String): Any? {
-        val normalisedMap = eventProperties.map { item ->
-            Utils.getNormalizedName(item.key) to item.value
-        }.toMap()
 
-        var value = normalisedMap[Utils.getNormalizedName(propertyName)]
+        var value = evaluateActualPropertyValue(propertyName)
 
         if (value == null) {
             value = when (propertyName) {
-                CLTAP_PROP_CAMPAIGN_ID -> eventProperties[NOTIFICATION_ID_TAG]
-                NOTIFICATION_ID_TAG -> eventProperties[CLTAP_PROP_CAMPAIGN_ID]
-                CLTAP_PROP_VARIANT -> eventProperties[INAPP_WZRK_PIVOT]
-                INAPP_WZRK_PIVOT -> eventProperties[CLTAP_PROP_VARIANT]
-                else -> systemPropToKey[propertyName]?.let { eventProperties[it] }
+                CLTAP_PROP_CAMPAIGN_ID -> evaluateActualPropertyValue(NOTIFICATION_ID_TAG)
+                NOTIFICATION_ID_TAG -> evaluateActualPropertyValue(CLTAP_PROP_CAMPAIGN_ID)
+                CLTAP_PROP_VARIANT -> evaluateActualPropertyValue(INAPP_WZRK_PIVOT)
+                INAPP_WZRK_PIVOT -> evaluateActualPropertyValue(CLTAP_PROP_VARIANT)
+                else -> systemPropToKey[propertyName]?.let { evaluateActualPropertyValue(it) }
             }
         }
 
         return value
     }
+
+    private fun evaluateActualPropertyValue(propertyName: String): Any? {
```

Comment:
Calling `getActualPropertyValue("propABC")` multiple times will re-compute same value multiple times. If possible can be memoized.

Reply:
it will be called for different `getActualPropertyValue `
The collision is a special case/really rare occurence.